libfwupdplugin/fu-volume.c

```diff
@@ -445,7 +445,16 @@ fu_volume_locker(FuVolume *self, GError **error)
 					 error);
 }
 
-/* private */
+/**
+ * fu_volume_new_from_mount_path:
+ * @mount_path: a mount path, typically a string
+ *
+ * Creates a volume with a specified mount path. Usually used internally or from the self tests.
```

Comment:
even though this is a "public" symbol, I want to discourage it's use.  How about if we kick it over to a header only to be used by self tests?  Or instead let the self tests set an environment variable and the daemon picks up that environment variable to use this symbol?

Reply:
I agree the export is ugly. Will have a rethink.

Reply:
I've pushed a fixup that adds back the env-variable. Maybe in the two self test cases it's actually fine.

Reply:
@superm1 do you like the export more or the env-var more?

Reply:
env variable, definitely.